src/Workspaces/Core/Portable/Execution/AssetBuilder.cs

```diff
@@ -75,6 +76,13 @@ public Asset Build(TextDocumentState state, SourceText text, CancellationToken c
             return asset;
         }
 
+        public Asset Build(OptionSet options, string language, CancellationToken cancellationToken)
+        {
+            // get around issue where this can't be captured in struct
+            var local = this;
```

Comment:
sure, we can do that way. did't go that arg path since I didn't want to add new method. none of other requires it.

I will change shape of tree node in next PR. so I will deal with in that PR.